File: ainpc-api/src/main/java/ro/ainpc/api/WorldAdminApi.java
package ro.ainpc.api;

import ro.ainpc.world.WorldNodeInfo;
import ro.ainpc.world.WorldPlaceInfo;
import ro.ainpc.world.WorldRegionInfo;
import ro.ainpc.world.WorldMode;

import java.util.Collection;

public interface WorldAdminApi {

    boolean isEnabled();

    WorldMode getWorldMode();

    Collection<WorldRegionInfo> getRegions();

    WorldRegionInfo getRegion(String regionId);

    WorldRegionInfo findRegion(String worldName, int x, int y, int z);

    Collection<WorldPlaceInfo> getPlaces();

    Collection<WorldPlaceInfo> getPlaces(String regionId);

    WorldPlaceInfo getPlace(String placeId);

    WorldPlaceInfo findPlace(String worldName, int x, int y, int z);

    Collection<WorldPlaceInfo> findPlacesByTag(String regionId, String tag);

    Collection<WorldNodeInfo> getNodes();

    Collection<WorldNodeInfo> getNodes(String regionId);

    Collection<WorldNodeInfo> getNodesForPlace(String placeId);

    WorldNodeInfo getNode(String nodeId);

    WorldNodeInfo findNode(String worldName, int x, int y, int z);

    Collection<WorldNodeInfo> findNodesNear(String worldName, double x, double y, double z, double radius, int limit);

    int getRegionCount();

    int getPlaceCount();

    int getNodeCount();
}

File: ainpc-api/bin/main/ro/ainpc/api/WorldAdminApi.kt
package ro.ainpc.api

import ro.ainpc.world.WorldMode
import ro.ainpc.world.WorldNodeInfo
import ro.ainpc.world.WorldPlaceInfo
import ro.ainpc.world.WorldRegionInfo

interface WorldAdminApi {
    val isEnabled: Boolean

    val worldMode: WorldMode

    val regions: Collection<WorldRegionInfo>

    val places: Collection<WorldPlaceInfo>

    val nodes: Collection<WorldNodeInfo>

    val regionCount: Int

    val placeCount: Int

    val nodeCount: Int

    fun getRegion(regionId: String?): WorldRegionInfo?

    fun findRegion(worldName: String?, x: Int, y: Int, z: Int): WorldRegionInfo?

    fun getPlaces(regionId: String?): Collection<WorldPlaceInfo>

    fun getPlace(placeId: String?): WorldPlaceInfo?

    fun findPlace(worldName: String?, x: Int, y: Int, z: Int): WorldPlaceInfo?

    fun findPlacesByTag(regionId: String?, tag: String?): Collection<WorldPlaceInfo>

    fun getNodes(regionId: String?): Collection<WorldNodeInfo>

    fun getNodesForPlace(placeId: String?): Collection<WorldNodeInfo>

    fun getNode(nodeId: String?): WorldNodeInfo?

    fun findNode(worldName: String?, x: Int, y: Int, z: Int): WorldNodeInfo?

    fun findNodesNear(worldName: String?, x: Double, y: Double, z: Double, radius: Double, limit: Int): Collection<WorldNodeInfo>
}
